### ISSUE TITLE
[Enhancement] Episode description should stretch across the whole width of the "Episode" columm

### ISSUE THREAD

**Issue by Marrib** [gpodder/gpodder#1122]
The episode description in the second line below the episode title is truncated at a certain width, not using the full width of the column. I think this should be changed to make the description as wide as the column is and truncate the description at the edge of the column. This would make a better use of the space, especially on larger monitors (like 1920x1080 etc.).

![truncation](https://user-images.githubusercontent.com/40357209/128597219-24b9dc9d-04c0-4448-9fd7-6c589d52499b.png)

**Comment by auouymous** [comment]
I have a narrow window and the label widget is clipping the title and description, and adding the three dots. I haven't looked at the code yet, but it appears in your case that gpodder is artificially clipping the text it stores in the label. It is possible this happens to reduce memory usage, because even though you only see one episode, the episode view widget contains every episode for every channel.

**Comment by Marrib** [comment]
> 
> 
> [...] even though you only see one episode, the episode view widget contains every episode for every channel.
Correct, I see only one episode because I have configured gPodder to hide deleted episodes in the list.

I really like gPodders small memory footprint and I would encourage the devs to keep it that way as far as this is possible.

However, with the amount of episodes that I currently have in gPodder (count 240, see screenshot above), gPodder uses 50,8 MB system memory according to the Windows Task Manager. I think this is very reasonable / nicely low. Surely, the memory usage would go up (a little?) if the description would fill the whole column width, but I doubt that the memory usage would rise extraordinarily. (Other podcast clients (like iTunes) use around 250-300 MB system memory just when running, even without having podcasts added or episodes downloaded. But please, don't ever make gPodder a memory hog like that).

Another suggestion would an option in the gPodder preferences to hide the episode description and to only show the title of the episodes in the episode list area.

**Comment by auouymous** [comment]
There is a function called `one_line_description()` that limits description to 120 characters, but linefeeds are converted to spaces, so it will show more than one line. Changing it to 1200000 had no impact on memory usage, which is odd because many youtubers like to post links to every episode they've ever made in the description of every episode.


> Another suggestion would be an option in the gPodder preferences to hide the episode description and to only show the title of the episodes in the episode list area.

That already exists in edit config under `ui.gtk.episode_list.descriptions`. The problem is it also hides the "from {channel}" that replaces description in the all episodes view. I'm not sure why, but my 1200 episodes use 112MB with that setting on or off, after a restart each time setting is changed.


> Alternatively, the second line could contain release date, duration and file size, which are separate columns as of now.

That would be nice, but you wouldn't know which channel each episode is from in all episodes.

**Comment by Marrib** [comment]
Editing the function `one_line_description()` somewhere in a config file may be not very convenient for the average user, so maybe there should be a setting in the preferences or maybe this should be default.

In the "All episodes" section this could be then solved by placing the following things in the second line: channel name, release date, duration and file size.
